### ISSUE TITLE
french model

### ISSUE THREAD

**Issue by vincauddu29** [syl22-00/pocketsphinx.js#88]
Bonjour, j'essaye de mettre en place la reconnaissance grâce au model Français en suivant [https://github.com/syl22-00/pocketsphinx.js-fr_FR-lium_french_f0](url) mais j'obtiens le message d'erreur :
Error in addWords with code [object Object]

mon code :
`
var initRecognizer = function() {

        data = {command: 'load', data: ["feat.params.js", "mdef.js", "means.js", "noisedict.js", "mixture_weights", "transition_matrices.js", "variances.js"].map(function(x) {return "../assets/vendor/recognizer/fr_FR/pocketsphinx.js-fr_FR-lium_french_f0/" + x;})}

        postRecognizerJob({command: 'initialize',
                            data: [["-hmm", "assets/vendor/recognize"],
                                ["-fwdflat", "no"],
                                ["-dict", "assets/vendor/recognizer/frenchWords62K.dic"],
                                ["-lm", "assets/vendor/recognizer/fr.lm.dmp"]]
            },
            function() {
                if (recorder) recorder.consumers = [recognizer];
                feedWords(wordList);});
    };`

Merci d'avance pour vos réponses.
Vincauddu29

**Comment by Thread7** [comment]
Loading via the .js files is an obsolete method.  The docs should really be updated.  Try the lazyLoad method.  Here is my working example with an English model.  
https://rawgit.com/Thread7/pocketsphinx.js.external/master/webapp/lazy.html
If you search/replace "cmusphinx-en-us-ptm-5.2" with the name of your model it might work.  But it may not too.  I have tried about 10 models and only one works with the lazyLoad feature.

**Comment by vincauddu29** [comment]
Hello, thanks for you answer, I have test your code with replace "cmu_sphinx-en-us-ptm-5.2" with "cmusphinx-fr-ptm-5.2" but i obtains 

"Error: Couldn't load ../cmusphinx-fr-ptm-5.2/feat.params. Status: 0" 

and

"uncaught exception: abort({}) at jsStackTrace@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:1:3106684
stackTrace@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:1:3106867
abort@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:19:5309
___syscall145@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:1:3469534
Gw@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:11:1
zx@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:11:1
by@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:10:1
fu@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:11:1
kn@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:9:1
yn@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:9:1
li@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:8:1
Li@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:8:1
Ki@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:8:1
Ji@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:8:1
GJ@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:12:1
dynCall_iii_24@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js line 1 > Function:2:12
constructor_body@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:1:3463419
__embind_register_class/</constructor<@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js:1:3457282
Recognizer@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/pocketsphinx.js line 1 > Function:2:29
initialize@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/recognizer.js:130:15
@file:///E:/wamp64/www/test/pocketsphinx.js-master/webapp/js/recognizer.js:58:2

If this abort() is unexpected, build with -s ASSERTIONS=1 which can give more information."

**Comment by Thread7** [comment]
Well it may be that your model files are not liked by lazyLoad for some reason.  The fix is beyond my abilities.  The only thing I could think is to use a different pocketsphinx.js file.  In my directory there is one named pocketsphinx-default.js.  Delete or rename the current one and then name the -default to pocketsphinx.js.  Then try again.

**Comment by vincauddu29** [comment]
thank you but i obtains the same error.

**Comment by Thread7** [comment]
Oh darn.  I don't know what to tell you.  I think most people run into a dead end with pocketsphinx.js and look to other options.  If you know other programming languages you could try one of the other CMUSPhinx based gitHub repos.

**Comment by vincauddu29** [comment]
yes but it's for a website, i go continue to search a solution. Again thank you for your answers and sorry for my english ;)

**Comment by syl22-00** [comment]
@Thread7 both lazyload and JavaScript compilation methods of using data files work and are documented. Also, using pocketsphinx.js only makes sense if it must run the the browser, otherwise, of course, a natively compiled version would work much faster.

@vincauddu29 can you put more details for your issue, and the steps you followed until you are blocked? Either you use the JavaScript compilation issues and use the provided compiled files or use the lazyload command and use the original binary files provided by CMU sphinx. I did try the French acoustic model compiled in JavaScript but it was too large to be usable. I am not sure it will be much better with lazyload. If possible, use a smaller model, if you find or can build one.

**Comment by Thread7** [comment]
@syl22-00 I agree that the methods are well documented.  Thanks for all your work on this.  But I don't think these methods work for a majority of models.  The model is either too large to compile or spawns an error like the one @vincauddu29 experienced when trying lazyLoad.  What models have been successful with either method?

**Comment by vincauddu29** [comment]
okay, sorry for the late.
@syl22-00 I have just followed your post https://github.com/syl22-00/pocketsphinx.js-fr_FR-lium_french_f0. What details do you want ?

@Thread7 I have try the worker method and your method with the french model

**Comment by nshmyrev** [comment]
cmusphinx-fr-5.2 is much better than lium-f0

**Comment by vincauddu29** [comment]
i have try also cmusphinx-fr-5.2 with the lazyload but not with syl22-00 method.

**Comment by JohnAReid** [comment]
>  I think most people run into a dead end with pocketsphinx.js and look to other options.

Would it be possible to put a complied French version of pocketsphinx.js on the site for downloading? It would enabled users to get up and running without the problems of the build system making it difficult.

**Comment by Thread7** [comment]
@JohnAReid that would be great but unfortunately many people have reported that compiling anything larger than the rm1_200 model into pocketsphinx.js doesn't work.  I think @vincauddu29 may be out of options since he has tried just about everything now.

**Comment by syl22-00** [comment]
@vincauddu29 this repo was part of an attempt I made a while ago to use models via bower packages. It did not go through and those packages are not up to date. Anyhow, I believe the best way to load models is to use the lazyload command and load raw files directly.

@JohnAReid compiling an acoustic model inside pocketsphinx.js is probably not a great idea. For the demo I added a small RM1 model inside, but just out of simplicity. One issue for instance is caching, any time you modify either a model or the code, the whole big compiled file will be re-downloaded instead of just the file that was modified. Also, as I just said, I think loading raw files is a better idea.

Another note is that large model files are difficult to handle memory-wise and will require more computing resources during recognition. The acoustic and language models provided by CMU Sphinx are as large as can be handled by a natively compiled application, and tend to be too large to be usable in `pocketsphinx.js`. So one better build (or adapt/prune from an existing one) models for his own need. It takes some effort but is is well documented on CMU Sphinx website.

**Comment by vincauddu29** [comment]
@syl22-00 ok but the raw files are [https://sourceforge.net/projects/cmusphinx/files/Acoustic%20and%20Language%20Models/French/?SetFreedomCookie](url) or I do compile them with your method ?

**Comment by JohnAReid** [comment]
@vincauddu29 did you ever get the French model to work - I am trying the cmusphinx-fr-5.2 model as suggested by nshmyrev - and I am getting the same error as you did - it seems as though the model is not be loaded as the words cannot see the phones:

Unknown phone bb in phone string bb

**Comment by batiot** [comment]
bb is from the english pronunciation dictionary use the cmu french one

example : https://github.com/batiot/pocketsphinx.js.external
